src/client/src/producer.rs

```diff
@@ -48,7 +59,32 @@ impl TopicProducer {
         K: Into<Vec<u8>>,
         V: Into<Vec<u8>>,
     {
-        self.send_all(Some((Some(key), value))).await?;
+        self.send_all(Some((Some(key.into()), value))).await?;
+        Ok(())
+    }
+
+    /// Sends a plain record with no key to this producer's Topic.
+    ///
+    /// The partition that the record will be sent to will be chosen in a round-robin fashion.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use fluvio::{TopicProducer, FluvioError};
+    /// # async fn example(producer: &TopicProducer) -> Result<(), FluvioError> {
+    /// producer.send("Key", "Value").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[instrument(
+        skip(self, value),
+        fields(topic = %self.topic),
+    )]
+    pub async fn send_keyless<V>(&self, value: V) -> Result<(), FluvioError>
```

Comment:
This is meant for the use case of: I want to send one record with no key. Previously, we would use `send_record`, but the problem with that is that it asked for an explicit partition number, which we no longer want to do. This is essentially the same function as `send_record` but minus the `partition` argument.

Reply:
let's not add more API unless it is absolutely necessary.

Reply:
Ok, so if we do not include this `send_keyless` API, then the only way to send a single record with no key is to do:

```rust
producer.send_all(Some((None, "Hello!"))).await?;
```

which I think is much more difficult to write and certainly not at all intuitive if you are a user searching for that functionality. If we do include `send_keyless`, users can use it to do the same thing:

```rust
producer.send_keyless("Hello!").await?;
```

So in this case I would actually argue that a function like `send_keyless` is actually very necessary. I am open to suggestions if you have any other preferences for the name of it though.

Reply:
Hmm. we still have `send` right? so it can be:
```
producer.send(None,"Hello!").await?;
```

Reply:
You cannot pass `None` to send. This is the API for `send`:

```rust
    pub async fn send<K, V>(&self, key: K, value: V) -> Result<(), FluvioError>
    where
        K: Into<Vec<u8>>,
        V: Into<Vec<u8>>,
    { ... }
```

So you must pass both a key and a value as arguments to send, you cannot use None. I have also tried other designs for this API but there does not seem to be a good solution for an optional key. This is the reason we need a separate API like `send_keyless`.

Reply:
Let's move creating new or updating existing API as another issue which doesn't belong here.

For handling none,  I suggest we use custom wrapper struct rather than using `Vec<u8>` directory.  This way we have full control of conversion.